### ISSUE TITLE
incorrect sitemaps with two domains in one installation

### ISSUE THREAD

**Issue by heimseiten** [MetaModels/core#1446]
## Checklist before I submit this issue report

I confirm that:
- [x] I have tested this with the latest version available
- [x] I have read documentation @ http://metamodels.readthedocs.org/en/ or http://metamodels.readthedocs.org/de/
- [x] I have checked the Contao community forums for references https://community.contao.org/
- [x] I have checked existing issues for duplicates and found none @ https://github.com/MetaModels/core/issues?q=is%3Aissue

## My environment is:
MetaModels 2.1 at a live website
MetaModels 2.2 at a test environment

| Key                                 | Value    | Comments                         |
| ----------------------------------- | ---------| ---------------------------------|
| PHP version:                        | 7.4 |                                  |
| Contao version:                     | 4.9 |                                  |
| MetaModels version:                 | 2.1 and 2.2  |                                  |
| Installed MetaModels packages:      | bundle_all |                                  |
| DCG version:                        | 2.1.x-dev and "dev-release/2.2.0": "2.2.x-dev"|                                  |

## Issue description

There are two domains in one Contao installation. The sitemap is only generated correct, for the domain which is logged in. For example:
When logged in in https://mmtest.ist-in-entwicklung.de/contao and press the sitemap generate checkbox then https://mmtest.ist-in-entwicklung.de/share/sitemap.xml is gernerated correct. But the sitemap from the other domain is incorrect:
https://mm.ist-in-entwicklung.de/share/sitemap1.xml

When generated with console "vendor/bin/contao-console contao:automator generateSitemap" both sitemaps are incorrect.

Looks like this:
```
<url>
<loc>https://mm.ist-in-entwicklung.de/https%3A/mm.ist-in-entwicklung.de/details/test1.html</loc>
</url>

```
## Screenshots
<img width="623" alt="Bildschirmfoto 2021-07-07 um 14 24 36" src="https://user-images.githubusercontent.com/6552484/124758734-2c9ee580-df2f-11eb-8c17-802d9a983a3c.png">
<img width="811" alt="Bildschirmfoto 2021-07-07 um 14 23 18" src="https://user-images.githubusercontent.com/6552484/124758742-2f013f80-df2f-11eb-9fad-8e07a63c38e6.png">

Can not update MM 2.2:
<img width="1296" alt="Bildschirmfoto 2021-07-07 um 14 15 05" src="https://user-images.githubusercontent.com/6552484/124759048-81426080-df2f-11eb-811d-fcca3d660e72.png">

**Comment by heimseiten** [comment]
￼Thank you very much. I'll send you an early adopter 2.2 request via email!

**Comment by heimseiten** [comment]
Now I have deleted the vendor folder and updated the Version 2.2, but the sitemaps are still incorrect.

`
<url>
<loc>https://mm.ist-in-entwicklung.de/https%3A/mm.ist-in-entwicklung.de/details/test1.html</loc>
</url>`

**Comment by zonky2** [comment]
o.k. - we ´ll check this

**Comment by zonky2** [comment]
o.k. - that is really the case :(
and if I change the (sub)domain and log in via it and create the XML, the links of my own (sub)domain are o.k. and the others are wrong...

**Comment by zonky2** [comment]
fixed in MM 2.2
